### PR TITLE
Finish implementing gdk::Atom

### PR DESCRIPTION
**Pull request by Susurrus**
I think this is all that's necessary?

### CODE REVIEW

**Comment by EPashkin** [comment]
Seems you need define `GlibPtrDefault` to fulfill need of https://github.com/gtk-rs/glib/blob/master/src/translate.rs#L967
```
impl GlibPtrDefault for Atom {
    type GlibType = *mut ffi::GdkAtom;
}
```

**Comment by Susurrus** [comment]
The trait bounds aren't satisfied even with the addition of a `GlibPtrDefault` impl for `Atom`. The compiler errors are complaining about a lack of this trait for `Vec<Atom>`, but I don't think that should actually be implemented. Is there something else missing here?

**Comment by EPashkin** [comment]
1. make `FromGlibPtrNone` and `FromGlibPtrFull` works from pointers not from value
```rust
//src/atom.rs
impl FromGlibPtrNone<*mut ffi::GdkAtom> for Atom {
    #[inline]
    unsafe fn from_glib_none(ptr: *mut ffi::GdkAtom) -> Atom { Atom(*ptr) }
}
```
2. Fix value atom errors
```rust
//src/event_owner_change.rs
     pub fn get_selection(&self) -> ::Atom {
        unsafe { from_glib_none(self.as_ref().selection as *mut _) }
     }
```
3. remove manual display.rs

**Comment by Susurrus** [comment]
I think this corrects the remaining issues.

File: src/auto/display.rs [hunk 1]
```diff
@@ -257,9 +260,11 @@ impl Display {
         }
     }
 
-    //pub fn store_clipboard(&self, clipboard_window: &Window, time_: u32, targets: /*Ignored*/&[&Atom], n_targets: i32) {
-    //    unsafe { TODO: call ffi::gdk_display_store_clipboard() }
-    //}
+    pub fn store_clipboard(&self, clipboard_window: &Window, time_: u32, targets: &[&Atom], n_targets: i32) {
+        unsafe {
+            ffi::gdk_display_store_clipboard(self.to_glib_none().0, clipboard_window.to_glib_none().0, time_, targets.to_glib_none().0, n_targets);
```
Comment (EPashkin) [inline src/auto/display.rs]: This function need tested with at minimum 2 element array.
Also about usage: not sure this or `targets: &[Atom]` better.

File: src/auto/device.rs [hunk 2]
```diff
@@ -206,9 +207,11 @@ impl Device {
         }
     }
 
-    //pub fn list_axes(&self) -> /*Ignored*/Vec<Atom> {
-    //    unsafe { TODO: call ffi::gdk_device_list_axes() }
-    //}
+    pub fn list_axes(&self) -> Vec<Atom> {
+        unsafe {
+            FromGlibPtrContainer::from_glib_container(ffi::gdk_device_list_axes(self.to_glib_none().0))
```
Comment (EPashkin) [inline src/auto/device.rs]: This function need be tested too.
For my default display it returns 2 devices and this function return empty vector, so I can't test it
```rust
let display = gdk::Display::get_default().unwrap();
let devices = display.list_devices();
println!("{}", devices.len());

for device in devices {
    println!("name: {:?}", device.get_name());
    let axes = device.list_axes();
    println!("{}", axes.len());
}
```

Comment (Susurrus) [inline src/auto/device.rs]: Indeed, I'm also getting 0's for `list_axes` for my two devices. Do you know what a `GdkDevice` is in practice, and output device or an input device? I can't glean much information from various internet searches.

File: src/auto/display.rs (hunk 1 above)
Comment (Susurrus) [inline src/auto/display.rs]: I think we want this to be `&[Atom]` since it's an input. The natural way to generate this will be to create a `Vec<Atom>` and pass a slice to the function. If the type is a slice of references, it's going to take an extra conversion to use this in the common case I think. It also shouldn't have the `target_length` argument, so I'll need to manually implement this.

**Comment by Susurrus** [comment]
Pushed a version that ignores the `store_clipboard` function for now and squash into 2 commits.

File: src/auto/display.rs (hunk 1 above)
Comment (EPashkin) [inline src/auto/display.rs]: Then `src/display.rs` need be restored with it

**Comment by Susurrus** [comment]
@EPashkin Is there anything else left to fix with this or are you waiting for gtk-rs/gtk#472 is in solid shape before you want to merge this?

**Comment by EPashkin** [comment]
As I don't understand what `gdk_display_store_clipboard` must do current version good for me and can be merged.
@GuillaumeGomez What you think about merging this now?

**Comment by GuillaumeGomez** [comment]
I didn't see anything blocking so if @EPashkin says ok, it's good to get merged. Thanks!
